### ISSUE TITLE
[HOLD for payment 2023-03-02] [$3000] mWeb/safari - Create Group button is hidden by keyboard when selecting participants

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11463]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/11309 

## Action Performed:
1. Login with any account
2. Tap on FAB icon
3. Select New Group
4. Type any email, or select any existing option
5. Verify that the green "Create group" button is not hidden by the keyboard instead of anchored above it

## Expected Result:
Create group button is not hidden

**Special note**: Reiterating for emphasis for the clarity of future contributors! We're only looking for proposals that will address upstream fixes to KeyboardAvoidingView. Thank you!

## Actual Result:
Create Group button is hidden when tap on selected user

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Mobile Web/safari

**Version Number:** 1.2.10.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/193172169-4295a539-bbb9-43a4-87c5-833117dd3e75.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by laurenreidexpensify** [comment]
Not overdue, reviewing this morning

**Comment by laurenreidexpensify** [comment]
https://www.upwork.com/jobs/~01948891954d6b32c7

**Comment by wildan-m** [comment]
In the video, the create group button is hidden when no participant is selected and then shown after one of the participants is selected.

Is that expected behavior?

Should we always show the "Create Group" button even when there is no selected participant?

**Comment by mananjadhav** [comment]
@flodnv @laurenreidexpensify Can you please confirm @wildan-m's comment above? Base on the code it does look like we're hiding the `Button` when we create the group. You can reproduce this even if you don't select the input text. Just scroll the list as soon as you click on the `Create Group` option.

```
const shouldShowFooter = (this.props.shouldShowConfirmButton || this.props.footerContent)
            && !(this.props.canSelectMultipleOptions && _.isEmpty(this.props.selectedOptions));

```

Second line of the condition in the above snippet.

**Comment by flodnv** [comment]
> the create group button is hidden when no participant is selected and then shown after one of the participants is selected.

>Is that expected behavior?

I don't know, but I would say yes...? 

>Should we always show the "Create Group" button even when there is no selected participant?

Why would we do that? Why would we show the `Create Group` button before you select anyone to add to the group? What would you expect to happen? 😕

**Comment by flodnv** [comment]
Oh, great point actually! @kbecciv why would you expect `Create group button is not hidden`?

**Comment by kbecciv** [comment]
@flodnv I'm expecting all environments worked the same, this is only my opinion. 

https://user-images.githubusercontent.com/93399543/196209810-a1d17d21-a3f0-49bc-960f-4f0cc9ddd3a3.mp4

**Comment by kbecciv** [comment]
Correct, attach the video for mWeb/Safari

https://user-images.githubusercontent.com/93399543/196234350-66e3c79f-ac36-475d-9422-072de190b1af.MP4

**Comment by wildan-m** [comment]
@kbecciv, (1) The button didn't show, or (2) it's covered by the keyboard? (We can find it when scrolling down)

@mananjadhav, @flodnv, if it's covered by the keyboard I think this issue is related to `KeyboardAvoidingView`, and most tasks related to it are on hold. Seems the internal team is working on it.

Similar issues can be found here: #10273

**Comment by laurenreidexpensify** [comment]
bountry increased to $1000

**Comment by mananjadhav** [comment]
@flodnv @kbecciv I am having a little problem in reproducing this. I did check the second video where the item was selected, but could this be that the button is hiding behind the keyboard?

Can we have a better reproduction steps with the complete video? I couldn't produce this on iPhone simulator and real device.

**Comment by flodnv** [comment]
>Can we have a better reproduction steps with the complete video?

Agreed.

I am still confused about https://github.com/Expensify/App/issues/11463#issuecomment-1281152039 -- only one account is selected, so by design, we don't show the button. 

Hummmm I tested on my Android app on my phone, and it does show the button when selecting only 1 participant 😬

**Comment by laurenreidexpensify** [comment]
@kbecciv can you please record a longer video with more detailed reproduction steps to help us confirm if this is still reproducible? thanks

**Comment by kbecciv** [comment]
Checking with team, will update here shortly

**Comment by kbecciv** [comment]
Issue is still reproduced in build 1.2.24.0
Steps:
1. Login with any account
2. Tap on FAB icon
3. Select New Group
4. Type any email
5. Hold your finger when cursor at the end
6. Verify that Create group button is not hidden

https://user-images.githubusercontent.com/93399543/200342288-a6c8bff4-3ede-4228-859e-4625879f33bd.mp4

**Comment by mananjadhav** [comment]
@kbecciv Pardon me for my ignorance, but I still couldn't make out if the button is hidden behind the keyboard?

**Comment by kbecciv** [comment]
@mananjadhav Correct, this PR https://github.com/Expensify/App/pull/11309 mention on step 6.  Verify that Create group button is not hidden. Which is Fail in mweb/safari

![image](https://user-images.githubusercontent.com/93399543/200596933-0859394e-2164-4723-91b5-c662124c9076.png)

**Comment by mananjadhav** [comment]
Okay thanks clears up the steps @kbecciv , but in your video are we scrolling till the end, that is now what I am able to see the `Create Group` button behind the keyboard. Here's the video.

https://user-images.githubusercontent.com/3069065/200636605-9ffd51a6-5b4e-4ce6-8235-7870510da55d.MP4

**Comment by flodnv** [comment]
@mananjadhav I think the suggestion is that the "Create group" button should not be hidden behind the keyboard, just like on a native Android app.

**Comment by flodnv** [comment]
>could this be that the button is hiding behind the keyboard? and is that what we plan to fix?

Yes!

**Comment by JmillsExpensify** [comment]
Removing Lauren since we're consolidating keyboard issues under one BZ. cc @tgolen

**Comment by mananjadhav** [comment]
> Verify that the green "Create group" button is not hidden by the keyboard instead of anchored above it

Thanks for the comments @flodnv @JmillsExpensify. This statement was critical to get aligned on the expected output.

---

Waiting for proposals.

**Comment by mananjadhav** [comment]
Waiting for proposals.

**Comment by hammad-6304** [comment]
**Proposal**
Hi hope you are doing great, I saw this job posting on Upwork.
I reproduce this issue on different browser eg Safari and chrome and successfully able to **resolve** this issue.

This issue occurs only on mobile browsers, React Web does not provide _KeyboardAvoidingView_ component by default so we have to create it manually.

Whenever keyboard pops-in in mobile browsers it triggers the change in **dimensions** of **View-Port**.

This issue can be fixed by sticking with the view-port dimensions so whenever it changes our View will adjust its height according to the new view-port.

I fixed this issue, I attached **screen recordings** just for reference of this **resolved issue** please review it once.

**Note**: It's working perfectly on the rest of platforms eg: Desktop, ios and android and also fixed on web now.

Looking forward to hear from your side,

Thanks.

https://user-images.githubusercontent.com/103407647/201782781-45c0c3ad-6f6e-4027-b9d8-86eae0b62505.MP4


https://user-images.githubusercontent.com/103407647/201782843-71e3811b-b3d4-46a1-96a3-9a7161cec97a.MP4

**Comment by hammad-6304** [comment]
### Proposal

###Root cause
Create Group Button is hidden by the on-screen keyboard on mobile browsers, Since React-Web doesn't provide ** KeyboardAvoidingView** by default so few components are hidden by the on-screen keyboard.

###Solution
Creating **KeyboardAvoidingView** manually for the web version to Lift-Up all the View behind the keyboard so it can be visible.
On-Screen Keyboard triggers the change in 'ViewPort' we can watch on this change and can lift-up the viewPort accordingly by applying suitable offset to make the button visible even if keyboard is open or closed on web platform.

Reference videos are attached of the fixed version of this issue on both **Safari** & **Chrome

https://user-images.githubusercontent.com/103407647/201874996-ce8a0503-b265-431a-8676-590c9df197c4.MP4

**.

https://user-images.githubusercontent.com/103407647/201874923-07f451ae-317f-406d-a276-7dae196175ba.MP4



cc: @JmillsExpensify @flodnv @kbecciv

**Comment by flodnv** [comment]
@mananjadhav what are your thoughts on this proposal?

**Comment by mananjadhav** [comment]
I agree with @hammad-6304 that it isn't a problem just with Create Group, but I can reproduce this on other screens as well.

Thanks for the proposal @hammad-6304.  The expected output looks to be correct.

But we use `react-native-web`, and afaik it does have `KeyboardAvoidingView`. So I think it would be better to fix any problem with that implementation? Does your solution fix that? or implements a new component altogether?

**Comment by hammad-6304** [comment]
@mananjadhav totally agree with your point `react-native-web` provide `KeyboardAvoidingView ` but, the problem is, it is not fully matured and context rich yet.
Only latest versions of `react-native-web` ships this feature still not fully functional but in `Mock State` (Reference is attached below) which will not be good enough to handle these scenarios on variety of browsers and devices since different browsers have different pre-processing & JS engines.


### Solution
Yes you are right @mananjadhav, The solution i proposed will handle all these scenarios and will be compatible not only with current issue (Create Group Button is hidden) but _**through out**_ the application.

It will be a new component handling all the scenarios for every possible web-based devices which will be totally dynamic and easy to consume anywhere in the codebase where needed. It'll be having its own default behaviours but will be totally customizable Via Props.

cc: @JmillsExpensify @flodnv 

<img width="1287" alt="Screenshot 2022-11-15 at 10 19 42 PM" src="https://user-images.githubusercontent.com/103407647/201989530-3b66221a-0942-4f3b-bccd-9f1d8c2a92a4.png">

**Comment by mananjadhav** [comment]
Thanks for the comment @hammad-6304. I am aware it's a mock state, but I was wondering if we should fix this upstream? or write a component ourselves. @flodnv do you have any opinion on this one? 

Could you elaborate on your proposal?  to share some technical details? code footprint we're looking to add? Also I hope you've gone through our contributing guidelines to ensure it fits into our style and coding guides.

**Comment by hammad-6304** [comment]
Yes, We can fix this upstream but there're a lot of chances that'll be bottle neck for us eg proper versioning, compatibility, bridging and linking through different versions or Base-Platform.
Since React-Native changed its architecture in the new version so we if we Fix this upStream we'll have to follow the new one so there'll be conflicts or either we need to upgrade our app with new one. So, by keeping in mind all the possible scenarios i think it'll be way better to go with a custom new component we'll be having strong grip and more control on it.

And yes Opinions matters most let's wait for the opinion from @flodnv.

### Technical Details

it'll be a custom `KeyboardAvoidingView` component.

- There'll be a separate component which will be responsible to re-adjust the size of page like KeyboardAvoidingView does.
- When on-screen keyboard opens up and close it triggers the change in Viewport dimensions, a Watcher will be implemented to watch these changes on real-time
- Since, this component will be only for web-based application so it'll look for the `Window Object` to determine the navigator, device and platform.
- This component will be used mostly on `full-screen non-scrollable` views so it'll mutate the dimensions by determining the current keyboard dimension and apply some offset.
- The Base component will be the `<View />` component and will return the `Children` (same a JSX element does).
- Can change Base component eg; if need to implement the same behaviour on `ScrollView` so can do on a quicker note.
- There'll be some Props which can be overloaded for achieving different behaviours and more behaviours can be added further later accordingly.


Yes i have gone through the contributing guidelines and also through the code base and am sure this solution is properly aligned with the guidelines, coding guides and performance meters.

**Comment by mananjadhav** [comment]
Thanks for the details.

We have listeners for view port like `withWindowDimensions`. We can use those. We can work out the kinks in the PR.  While we do maintain our own forks for react-native and so on, I am fine with us implementing a component here.

@flodnv What are your thoughts.  I am more or less convinced with @hammad-6304's approach here.

🎀 👀 🎀 
C+ reviewed

**Comment by hammad-6304** [comment]
Thanks for the responses. 

Looking forward.

**Comment by Julesssss** [comment]
I'm going to hold this one temporarily, as we already have spent some time working on a custom KeyboardAvoidingView component internally. Thanks for your patience here.

CC @tgolen for when you return from OOO

**Comment by hammad-6304** [comment]
@Julesssss Thanks for the update.

Can you please confirm me, is there any further actions for me on this proposal and solution? 
What should i have to do next?
Can i have an invitation to be the part of community eg: Slack internal groups and stackoverflow community?

Thanks, eagerly waiting for the humble response from your side. 

CC @mananjadhav  @flodnv @JmillsExpensify

**Comment by tgolen** [comment]
I've read through this issue and taken some of the proposals into account, but I do not think we should fix this in our codebase. We should absolutely be fixing this in the upstream repos and it will not only make the react-native-web lib better for us, but for everyone else using it. This bug isn't so bad that we need any kind of temporary hack or work around in our codebase.

**Comment by flodnv** [comment]
Agreed, we will only accept proposals that fix this upstream. Taking this off hold. Thanks!

**Comment by JmillsExpensify** [comment]
Reiterating for emphasis for the clarity of future contributors! We're only looking for proposals that will address upstream fixes to `KeyboardAvoidingView`. Thank you!

**Comment by mananjadhav** [comment]
@JmillsExpensify Can we also updated the requirement in the GH body? Just in case someone misses in the comments.

**Comment by JmillsExpensify** [comment]
Sure thing. Done!

**Comment by flodnv** [comment]
Maybe this should HOLD on https://github.com/Expensify/App/issues/10670

**Comment by tgolen** [comment]
Ah, I think it's the other way around. #10670 needs to HELD for this one.

**Comment by flodnv** [comment]
Ah ok, sure, I did it that way because https://github.com/Expensify/App/issues/10670 had more proposals/activity, but we can keep this one. Seems like next step is getting https://github.com/necolas/react-native-web/issues/2430 resolved?

**Comment by flodnv** [comment]
Discussing this here: https://expensify.slack.com/archives/C01GTK53T8Q/p1669192375616369?thread_ts=1669139046.925109&cid=C01GTK53T8Q

@rawalyogendra do you think you can submit that PR today? 🙏

**Comment by flodnv** [comment]
>I have the PR ready.

Awesome! Can you link to it here? I am not finding it. 

----

> I was actually testing the changes with our app and I think we have to do some changes on our side too.

Alright, let's get your react PR merged first. @mananjadhav how do these changes sound to you?

**Comment by mananjadhav** [comment]
@rawalyogendra Can you confirm if the changes are based on the latest `main` branch? I remember 1 or 2 PRs changing the behavior for card styles. Want to check your recommended changes are still applicable for the current latest code.

**Comment by flodnv** [comment]
Thanks for the PR @rawalyogendra ! Let's fix the comment and hopefully then it gets merged? 🙏 https://github.com/necolas/react-native-web/pull/2438#issuecomment-1325575727

**Comment by rawalyogendra** [comment]
yeah I will do this today. I will also look for a elegant solution for our application as well.

**Comment by JmillsExpensify** [comment]
Still working to get the related PR merged.

**Comment by flodnv** [comment]
Still waiting on that PR, posted in Slack about it... https://expensify.slack.com/archives/C01GTK53T8Q/p1669800759208819?thread_ts=1669673848.122059&cid=C01GTK53T8Q

**Comment by JmillsExpensify** [comment]
Alright, I was just taking stock of this issue. The only Contributor helping solve this issue upstream. However, @rawalyogendra isn't assigned to this issue. cc @flodnv @mananjadhav

**Comment by mananjadhav** [comment]
> However, @rawalyogendra isn't assigned to this issue 

I think we can assign @rawalyogendra  

@rawalyogendra, meanwhile I think we can add a proposal to fix it for our app? and review it?

**Comment by rawalyogendra** [comment]
Working on it @mananjadhav

**Comment by rawalyogendra** [comment]
## Proposal
### Problem
As we know, on iOS when a keyboard is opened/closed, the viewport height is not updated. Currently, we depend upon viewport height to create our `KeyboardAvoidingView` on mWeb.

### Solution
We can determine the `height` of the `window` by using the `windowHeight` provided by `withWindowDimension`

Replace this:
https://github.com/Expensify/App/blob/2e8d90736a19de0bf0332fb36afc42bab074434f/src/components/ScreenWrapper/index.js#L92-L93

With:
```js
<KeyboardAvoidingView style={[styles.w100, styles.h100, {maxHeight: this.props.windowHeight}]} behavior={this.props.keyboardAvoidingViewBehavior}>
```

After applying changes:

https://user-images.githubusercontent.com/7548730/205505908-edd7b3cc-246c-4a3b-892b-89cca36d6de1.mp4

Also fixes #10670 

This works after using this fix for `react-native-web`
https://github.com/necolas/react-native-web/pull/2438

**Comment by mananjadhav** [comment]
Thanks for the proposal @rawalyogendra. Two quick questions:

1. Does this have any impact on Android mWeb or other platforms?
2. Can the transition when the keyboard hides be a little smooth?

Rest it looks fine.

**Comment by rawalyogendra** [comment]
@mananjadhav I looked into the transition (animation) before submitting the proposal. Unfortunately, iOS only updates the viewport height after the keyboard animation is completed, so the button's position is updated after the keyboard is completely opened/closed which is why it looks a bit laggy.

I haven't tested all other platforms/devices yet but Android mWeb works without any issues.

I will get back to you after I test this on other platforms.

**Comment by mananjadhav** [comment]
The proposal looks fine to me @rawalyogendra. But with the upstream PR stuck, I am not sure what's the next step here.

@flodnv @JmillsExpensify It's been more than 2 weeks that the PR is stuck. What do you think we should do here?

**Comment by rawalyogendra** [comment]
@mananjadhav From what I understand, we maintain our own fork of `react-native-web`, so can't we just add these changes onto our own fork?

**Comment by flodnv** [comment]
Discussing in Slack: https://expensify.slack.com/archives/C01GTK53T8Q/p1670507532249869?thread_ts=1669673848.122059&cid=C01GTK53T8Q

Let's wait until Monday and then go the fork route. Thank you @mananjadhav @rawalyogendra for your continued work on this 🙇

**Comment by mananjadhav** [comment]
@rawalyogendra I think the maintainer has requested to add unit tests. Can we please add that to move this forward?

**Comment by mananjadhav** [comment]
@rawalyogendra do we have an ETA on tests for the upstream PR?

**Comment by JmillsExpensify** [comment]
It's also officially Monday, so if we don't have an ETA then let's go the fork route as discussed. 

P.S. @flodnv I'm OOO this week, so would appreciate if you could help move it forward while I'm out.

**Comment by flodnv** [comment]
@rawalyogendra how are these tests going?

@JmillsExpensify I am going ooo until January, I am leaving this one to you until then 🙇

**Comment by rawalyogendra** [comment]
@flodnv I will push my changes today. So that means this will be handled in January.

**Comment by mananjadhav** [comment]
I think we can have someone else to look at in case @flodnv @JmillsExpensify are away. I'll be there @rawalyogendra.

**Comment by flodnv** [comment]
Yes, everyone else but me will help push forward. @JmillsExpensify will recruit another internal engineer if/when needed 🙃 It would be a great new year's present if I come back and this issue is fixed 😁 

Thank you @rawalyogendra @mananjadhav

**Comment by mananjadhav** [comment]
@rawalyogendra did you get a chance to update the tests for the upstream PR?

**Comment by JmillsExpensify** [comment]
@rawalyogendra I wanted to check in on @mananjadhav's comment above?

**Comment by rawalyogendra** [comment]
@mananjadhav sorry this is taking a bit longer than expected. I have hit a dead end and working on resolving it.

If you have any ideas on writing tests for my changes on upstream or point me in a right direction, it would be very much appreciated.

**Comment by rawalyogendra** [comment]
@JmillsExpensify @mananjadhav I have added the tests that was requested by upstream repo owner.

Waiting for the review, will continue working on this.

Thank you for your patience and support.

**Comment by JmillsExpensify** [comment]
Thank you! Really appreciate us helping close the loop.

**Comment by JmillsExpensify** [comment]
@rawalyogendra We've been discussing this in our open source [Slack community](https://expensify.slack.com/archives/C01GTK53T8Q/p1671463612134789), so I'm including the link for reference. Long story short, we've decided to push this fix in our own Fork to keep this moving forward. That way we can wrap up the initial phase of this project and then still work on getting the upstream fix merged longer-term. Thoughts on that approach?

**Comment by rawalyogendra** [comment]
@JmillsExpensify We discussed about that possibility in this thread as well.

I think it is the logical step now as it will take some time to get the upstream PR merged. I will keep an eye on any change requests in the upstream PR.

**Comment by rawalyogendra** [comment]
@JmillsExpensify will the team be working on the PR internally or should I open one on our fork?

**Comment by mananjadhav** [comment]
@rawalyogendra Forked PR will be coded by you, but we'll have someone internally review it.

**Comment by rawalyogendra** [comment]
@mananjadhav okay thanks. I will be waiting for further instructions ☺️

**Comment by JmillsExpensify** [comment]
Pulled in @youssef-lr given that Florent is out for this week and the next. Youssef - Can you help @rawalyogendra with next steps?

**Comment by youssef-lr** [comment]
@rawalyogendra The fork is over here: https://github.com/Expensify/react-native-web, feel free to raise a PR there and link it here!

**Comment by youssef-lr** [comment]
Hey @rawalyogendra! You will need to fork the upstream repo https://github.com/necolas/react-native-web then submit a PR from _your_ fork to our fork over here: https://github.com/Expensify/react-native-web.

**Comment by rawalyogendra** [comment]
@youssef-lr I have raised the PR. Please take a look and let me know if you find any issues.

**Comment by JmillsExpensify** [comment]
Added @youssef-lr as reviewer for the PR since you're jumping in while Florent is out on vacation. Then I'm also adding a reminder for myself that we need to create a separate issue once our fork is updated, so that when the upstream fix is eventually merged, we can point to that instead of our fork.

**Comment by JmillsExpensify** [comment]
Working through the linked PR. Special note: If we're able to merge it in the next 2 business days, I think we should award @rawalyogendra a 50% bonus for his work. cc @flodnv @youssef-lr

**Comment by rawalyogendra** [comment]
@JmillsExpensify thank you. I will be available for any change requests to the relevant PRs

**Comment by mananjadhav** [comment]
@youssef-lr Do you have any comments on the PR? I see you've raised the same comment about smooth transition and we don't seem to have a solution for this.

**Comment by youssef-lr** [comment]
@mananjadhav I'm not sure that transition is something we can live with. @rawalyogendra do you happen to know why the transition on native is much smoother?

**Comment by rawalyogendra** [comment]
@youssef-lr I am not sure why the transition is smooth on native devices.

As far as I can tell there is no other way to do this except using `visualViewport` which is what we are using right now on `react-native-web`.

In my opinion, it would be preferable for the button to display with a slight delay rather than not displaying at all.

I am actively seeking out potential solutions to this issue and will happily share any updates with you as soon as I find a more effective solution

**Comment by JmillsExpensify** [comment]
@youssef-lr I think you're still around this week? Do you mind circling back on this issue to help close the loop on the outstanding PR and @rawalyogendra's comment above?

**Comment by youssef-lr** [comment]
@JmillsExpensify I'm going OOO as soon as I wrap up some of my issues including this one!

**Comment by youssef-lr** [comment]
I'm not a fan of this transition to be honest. @JmillsExpensify can we live with this? Perhaps we can reimagine the design to not rely on the button being hidden by the keyboard? Or investigate more if this can be solved some other way.

https://user-images.githubusercontent.com/9680864/210099452-840c27b4-7600-4df1-bff9-97a06953e347.MP4

**Comment by youssef-lr** [comment]
Started a discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1672424442428259).

**Comment by youssef-lr** [comment]
@rawalyogendra As per the thread above, we have agreed on the animation needing work before we can ship this out, would you be able to look into it? Thanks for your great work so far!

**Comment by rawalyogendra** [comment]
@youssef-lr I will try to fix the animation

**Comment by flodnv** [comment]
Thanks @rawalyogendra !

**Comment by flodnv** [comment]
@rawalyogendra do you have any ETA for an update? 🙇

**Comment by JmillsExpensify** [comment]
Just my opinion, but I think we've surpasses the original scope of this issue and now need to re-group. I would recommend that we: 

 - Close this issue, as we've pushed an update to the fork and also submitted a PR for review upstream
 - Compensate @rawalyogendra for his work thus far
 - Create a new issue for improving the transition or potentially explore not having the button be hidden by the keyboard.

**Comment by JmillsExpensify** [comment]
Quick update after a 1:1 discussion with @flodnv . He reminded me that @rawalyogendra has pushed a PR for the fork, but it's still not approved and merged. Accordingly, @flodnv is going to jump in and review, since that solves the bug for which this issue was created. Then we'll create a fresh issue to address the transition and potentially explore not having the button be hidden by the keyboard. 

Accordingly, I'm removing the HOLD and re-adding the reviewing label.

**Comment by mananjadhav** [comment]
@JmillsExpensify quick questions,

1. @flodnv is going to review the forked PR? But the transition isn't smooth with that change. Yet we will still merge the PR?
2. Any specific reason we're not using this issue? Technically the goal is still the same `Create Group button is hidden by keyboard`. 
3. We have this type of buttons at multiple places, Forms, Payments Screen, etc. so is the new issue going to track all the buttons or just Create Group/IOU ones?

**Comment by flodnv** [comment]
How is the transition on platforms other than mweb/Safari? If it's only not smooth on mweb/Safari, I think we just go with what we got now.

**Comment by JmillsExpensify** [comment]
Great questions. As for the transition, I agree with @flodnv that if it's just mWeb and Safari, we should go with what we have. 

>Any specific reason we're not using this issue? Technically the goal is still the same Create Group button is hidden by keyboard.

The main motivation is that I think we've reached a fair stopping point for this contributor from a scope perspective, and then related, so yes, the plan would be to tackle this more holistically as well.

**Comment by flodnv** [comment]
Please let us know your thoughts @mananjadhav 🙇

**Comment by mananjadhav** [comment]
I agree with what we're planning. On the other platforms it works fine, so only mWeb Safari is the problem. I think we should revisit our bottom docked buttons overall.

**Comment by flodnv** [comment]
Ok so,

>@flodnv is going to review the forked PR? But the transition isn't smooth with that change. Yet we will still merge the PR?

Yes, because the transition is only not smooth on mweb/safari, which is a small subset of users. 

>Any specific reason we're not using this issue? 

So that we can close out this old 120+ comments issue and start with a fresh one 

>We have this type of buttons at multiple places, Forms, Payments Screen, etc. so is the new issue going to track all the buttons or just Create Group/IOU ones?
> I think we should revisit our bottom docked buttons overall.

I am not really following what you mean on this, can you expand please?

**Comment by mananjadhav** [comment]
What I meant is we've got bottom-docked buttons across multiple screens - say Payments, IOU, etc. So from the design side  are we revisiting that the buttons shouldn't be docked at the bottom so that this issue doesn't occur? or we're basically going to fix the bottom dock behavior.

**Comment by JmillsExpensify** [comment]
Ah yeah, you're right that we've talked about this in a couple of Github issues and in our Slack community. At the moment, it's filed as a "potential improvement" but I don't think we have an immediate plans. In fact, we're about to kick off a refactor of IOUs, so we can revisit then.

**Comment by JmillsExpensify** [comment]
So all in all, I think what we've landed on is acceptable. Are we aligned on these are the next steps?

 - **Issue payment**. I would like to recommend that we issue @rawalyogendra a 50% bonus for this issue, or $3,000, primarily due to responsiveness and flexibility. Same goes for @mananjadhav as C+. 
 - **Create another issue**. We'd put it on hold, but minimally we should document what isn't ideal about where we left off, so we can circle back, potentially moving to bottom-docked button to be top-docked. And so on and so forth. 

Does that sound good to everyone?

**Comment by mananjadhav** [comment]
Sounds good. Much appreciated

**Comment by rawalyogendra** [comment]
Sounds Good @JmillsExpensify.

Thank you so much.

**Comment by JmillsExpensify** [comment]
Cool. Upwork job [is here](https://www.upwork.com/jobs/~01a2b489bfe7b20e8e). Can both of you apply, so we can get that process started? I'll issue the bonus on final payment.

**Comment by rawalyogendra** [comment]
I have applied to the job. Thanks again @JmillsExpensify

**Comment by mananjadhav** [comment]
Applied @JmillsExpensify

**Comment by flodnv** [comment]
Thanks everyone. I left a comment on [the PR](https://github.com/Expensify/react-native-web/pull/10), but I don't feel qualified to properly review it, so I asked in Slack for more eyes: https://expensify.slack.com/archives/C02NK2DQWUX/p1673613670382889

**Comment by mananjadhav** [comment]
@flodnv I've reviewed the PR along with the others.

**Comment by mananjadhav** [comment]
@flodnv Any blockers on merging the PR?

**Comment by flodnv** [comment]
Yes, I left a couple of comments just now.

**Comment by JmillsExpensify** [comment]
Everyone has been hired in Upwork.

**Comment by flodnv** [comment]
https://github.com/Expensify/react-native-web/pull/10 is merged, onto the App PR now! Almost there! 🙏

**Comment by JmillsExpensify** [comment]
Actually, I just realized that I got mixed up @flodnv. Do you mind linking the App PR for clarity?

**Comment by rawalyogendra** [comment]
@JmillsExpensify I am working on it. Will be up by today

**Comment by flodnv** [comment]
Whoa, the upstream change has been merged! https://github.com/necolas/react-native-web/pull/2438#event-8358765305

Now we need to wait for one day for it to be released... We'll see! We're tracking internally the fact that we need to merge the upstream back into our repo.

**Comment by flodnv** [comment]
@mananjadhav please give the PR a final review with the checklist when you get a chance 🙇

**Comment by JmillsExpensify** [comment]
Woah, that's super cool! Yeah let's close the loop on what's left on our side. 

> We're tracking internally the fact that we need to merge the upstream back into our repo.

Yes, [that's here](https://github.com/Expensify/App/issues/13846). I'm taking it off hold now.

**Comment by mananjadhav** [comment]
@flodnv Apologies I couldn't check it earlier. I've tested and completed the checklist.

**Comment by JmillsExpensify** [comment]
Linked PR hit staging today, so we should kick off the regression period pretty soon here.

**Comment by JmillsExpensify** [comment]
Nice, we've got some time to get this checklist squared away, though @mananjadhav and @flodnv, you guys want to get us kicked off?

**Comment by mananjadhav** [comment]
@JmillsExpensify @flodnv I did think about the checklist and I am not sure if we have any offending PR here. The fix was added to an upstream repo react-native-web and not fixed by changing any code from our previous PRs?

**Comment by flodnv** [comment]
I'm a bit out of the loop on this one. @mananjadhav @rawalyogendra I think I've seen we're working on regressions?

**Comment by JmillsExpensify** [comment]
Holding on payment until we clear up where things are currently at.

**Comment by rawalyogendra** [comment]
@flodnv I have added my proposal here https://github.com/Expensify/App/issues/14716#issuecomment-1419964610 Still waiting for someone to look into it.

**Comment by mananjadhav** [comment]
We're working on the regression, but I don't think we have a concrete solution yet. I think @parasharrajat did review the solution and I agree with him that we should first revert this change, atleast on the App side. We keep this open and continue with the solution for this as well as both the linked regressions here.

What do you folks think? @rawalyogendra @JmillsExpensify @flodnv ?

**Comment by rawalyogendra** [comment]
@mananjadhav please look into the comment above. The solution linked with that comment actually fixes most of the issues

**Comment by flodnv** [comment]
Thanks @rawalyogendra, commented there that your proposal looks good to me. Let's see what others think.

**Comment by mananjadhav** [comment]
Posted my comment there.

**Comment by flodnv** [comment]
https://github.com/Expensify/App/pull/15100 is in final review, thanks @mananjadhav

**Comment by JmillsExpensify** [comment]
That title wasn't really accurate for updating for seven days from today so this doesn't get lost.

**Comment by JmillsExpensify** [comment]
Looks like the linked PR is still in review, so I'm going to update the title for another 7 days out so it doesn't get lost in the shuffle.

**Comment by flodnv** [comment]
The fix was deployed to staging yesterday, hopefully we can issue final payment next week! 🙏

**Comment by JmillsExpensify** [comment]
Also I'm going to be OOO next week, so @trjExpensify can you circle back on this one then and issue payment?

**Comment by JmillsExpensify** [comment]
@flodnv given that we had a regression in play, are we effectively at C/C+ payment of $1,000?

**Comment by trjExpensify** [comment]
>Also I'm going to be OOO next week, so @trjExpensify can you circle back on this one then and issue payment?

Yep, assigning myself.

**Comment by MelvinBot** [comment]
⚠️ Could not update price automatically because there is no linked Upwork Job ID. The BZ team member will need to update the price manually in Upwork.

**Comment by flodnv** [comment]
>@flodnv given that we had a regression in play, are we effectively at C/C+ payment of $1,000?

We had already agreed to a $3,000 payment: https://github.com/Expensify/App/issues/11463#issuecomment-1380485054. Title updated for clarity. 

Given this was a very difficult task and a very difficult regression that nobody could've caught, I'd recommend only a third penalty (or a further increase in the original payout, which is the same result), so let's say the payout will be $2,000. Are you happy with that @rawalyogendra @mananjadhav ?

**Comment by redstar504** [comment]
It looks like this issue might have something to do with https://github.com/Expensify/App/issues/15543.

**Comment by trjExpensify** [comment]
Awaiting feed back on https://github.com/Expensify/App/issues/11463#issuecomment-1446273558

**Comment by mananjadhav** [comment]
Yes I think this would work. Thanks for the consideration @flodnv

**Comment by flodnv** [comment]
@rawalyogendra could you please check out https://github.com/Expensify/App/issues/15543 ? 😬

**Comment by trjExpensify** [comment]
Seems like we're still waiting here on @rawalyogendra input.

**Comment by rawalyogendra** [comment]
Hey Everyone,

I've been working on this issue for the past few days, but unfortunately, I couldn't find a fix for it. The problem lies in the fact that the VisualViewport `resize` event is only triggered after the On-Screen Keyboard(OSK) animation is completed.
<details><summary>Root Cause</summary>

https://user-images.githubusercontent.com/7548730/223794405-2d100184-bcb6-4e75-aadf-7151f0bd0071.mov

</details>

If we decide to revert the original PR and subsequent regression fixes, we should still retain some of the changes, such as 

https://github.com/Expensify/App/blob/fb2014493988d2f3d30f1bcae160c1d307b5da7e/web/index.html#L80

This change will maintain the old behavior on our mWeb(Android) platform, which was recently changed (you can read about it here: https://developer.chrome.com/blog/viewport-resize-behavior/). This new change in Chrome caused similar OSK-related issues on our mWeb(Android) platform too.

https://github.com/Expensify/App/blob/fb2014493988d2f3d30f1bcae160c1d307b5da7e/web/index.html#L52-L60

This change will fix the issue of mWeb(iOS) scrolling a webpage when there's a text input at the bottom of the page, which was pushing the page title outside the viewport.
<details><summary>Before</summary>

https://user-images.githubusercontent.com/7548730/223795200-3030414e-fdd0-4cea-9c2f-7ad3e830d19d.mov

</details>
<details><summary>After</summary>

https://user-images.githubusercontent.com/7548730/223795539-7208d8f1-7051-46e1-9a7b-4b444238755d.mov

</details>

In my opinion, we should open up this issue to other contributors and let them have a say on how we proceed. I'm okay with whatever decision you guys make.

Also, just wanted to let you all know that I'm available for any questions or help in fixing this issue. Feel free to reach out.

Thank you all for your immense patience. To be honest, diving deep into the rabbit hole of Safari bugs has been a fun adventure for a web developer like me!

**Comment by JmillsExpensify** [comment]
Ok coming back into this issue. Thanks @trjExpensify for carrying this forward while I was away.

**Comment by mananjadhav** [comment]
I agree with reopening #15543 to others and try to see if it can be resolved. Current changes have anyway resolved a couple of issues and we should retain them.
